handlers/reader.go

```diff
@@ -83,3 +84,23 @@ func getEnvProcess(envVars []string) string {
 
 	return value
 }
+
+func buildLabelsAndAnnotations(dockerLabels map[string]string) (labels map[string]string, annotations map[string]string) {
+	for k, v := range dockerLabels {
+		if strings.HasPrefix(k, annotationLabelPrefix) {
+			if annotations == nil {
```

Comment:
only initialize map if we have at least one annotation, so nil is returned otherwise